### ISSUE TITLE
Restart Working/Break cycle from the beginning

### ISSUE THREAD

**Issue by andrey2409gmailcom** [rcaelers/workrave#335]
I cannot restart the timer to start working cycle from the beginning
Sometimes I take a break from work in the middle of the cycle for a while, enough time to rest.
And after that I need to start the working cycle from the beginning, but I can't.

I saw this feature in another application (browser extension), and this feature was very helpful and usefull
https://github.com/schmich/marinara

In that application, the buttons are named like this:
Restart Timer
      - Start Focusing (working cycle)
      - Start Short Break
      - Start Long Break

**Comment by Chealer** [comment]
@andrey0924: This is an ITS. As such, each ticket should report an issue, not simply make a request. Please describe the issue you experience, and specify which Workrave version you use when it occurs.

**Comment by kees-jan** [comment]
@Chealer Not sure if this is the right place for this discussion, but in spite of being a professional software engineer, the abbreviation "ITS" was new for me, and googling its (pun not intended) meaning was nontrivial.

Also, I feel that making (feature) requests is perfectly reasonable here, and this one is clear enough as it is. Frankly, a missing or desired feature is not an issue. "issue" would indicate that something is wrong, and not being complete (yet) or discussions about in which direction Workrave should even go should have their place somewhere, and are not necessarily issues.

Lastly, I've noticed that you are a bit brief, possibly due to the volume of comments you are making. However, when reading them in isolation (as our users are likely to do), they sometimes come across (at least to me) as a bit unhelpful and unwelcoming.

Just my 2cts, and again my apologies if this is the wrong place.

**Comment by kees-jan** [comment]
@andrey0924 If your break was long enough to rest, and yet the cycle hasn't reset automatically, then maybe your break duration is set too high?

If your screen was locked during your break, then there should be a (natural) restbreak window visible when you return. If you press the "skip" button on it, the cycle should reset.

**Comment by Chealer** [comment]
Hi @kees-jan,

> [@Chealer](https://github.com/Chealer) Not sure if this is the right place for this discussion, but in spite of being a professional software engineer, the abbreviation "ITS" was new for me, and googling its (pun not intended) meaning was nontrivial.

Hehehe
Search engines are generally not the best tool to look up an acronym, in particular when its sequence of letters is the same as a common word; although Wiktionary is far from reliable, it is reasonably comprehensive and would have simplified your research [in this case](https://en.wiktionary.org/wiki/ITS).
 
> Also, I feel that making (feature) requests is perfectly reasonable here, and this one is clear enough as it is. Frankly, a missing or desired feature is not an issue.

I certainly did not claim that making requests here was unreasonable. This topic is non-trivial, but at least in general, yes, the lack of a feature does constitute an issue. For example, the lack of revision history in the wiki of Microsoft Teams constitutes an issue.

> "issue" would indicate that something is wrong,

"issue" is hypersemous, but at least in the context of issue tracking, not necessarily. For example, the absence of an elevator in a house constitutes an issue to some (like disabled people), yet there is nothing _wrong_ with a house lacking an elevator.

> and not being complete (yet) or discussions about in which direction Workrave should even go should have their place somewhere, and are not necessarily issues.

Right, but general discussions belong in a forum, not in an ITS. I am afraid you are right that Workrave has no forum for that topic... but that is [another issue](https://github.com/jquery-validation/jquery-validation/issues/2198).

To clarify, my comment was not that this request was inappropriate on this ITS; only that its phrasing/presentation was not. Simon Tatham [explains it with a great analogy](https://www.chiark.greenend.org.uk/~sgtatham/bugs.html#symptoms). I am guessing from what you and Andrei wrote that my understanding of his issue corresponds to yours, but we should not have to speculate.
 
> Lastly, I've noticed that you are a bit brief, possibly due to the volume of comments you are making.

Many people emphasize the importance of adapting one's communication style to one's interlocutors. I figure someone brief enough to omit specifying the version of the software they use when reporting one of its issues calls for a brief enough style.

> However, when reading them in isolation (as our users are likely to do), they sometimes come across (at least to me) as a bit unhelpful and unwelcoming.

Considering that I was the first one to thank the submitters of at least tickets #483, #588, #221, #539, #551, #584, #329 and #527 for these, if you think _I_ come across as unwelcoming, it would most likely be best to discuss this project's unwelcomingness in a larger forum.

As for coming across as unhelpful, if some consider any comment unhelpful, I guess commenting on that comment would have better chances of becoming helpful.
 
> Just my 2cts, and again my apologies if this is the wrong place.

Some of your comment is not at the ideal place, but as I am not aware of a better place at this time, I don't take offense. Thanks

**Comment by Chealer** [comment]
> If your screen was locked during your break, then there should be a (natural) restbreak window visible when you return.

That depends on how Workrave is configured, and I believe that preference is disabled by default.

**Comment by kees-jan** [comment]
Hi @Chealer 

Judging form your response, you are a fellow software engineer. Welcome. Most people using workrave are not, and sometimes the communication styles of software and non-software people collide. That was the point I was trying to make. In your reply you argue that your message was "correct" (for some definition of the word). That suggests to me that you may have missed my original point (my apologies, maybe I should have been clearer). And now, ironically, your communication style collided with mine 😉 .

**Comment by Chealer** [comment]
Hi @kees-jan,

> Judging form your response, you are a fellow software engineer.

I am indeed a senior software developer.

> Most people using workrave are not, and sometimes the communication styles of software and non-software people collide. That was the point I was trying to make.

If that was your point, I am sorry to say that was quite unclear for me, although I consider I have extensive inbound communication style compatibility.

You did point out that 1 acronym I used was little known and non-trivial to look up, but ITS acronyms have been in wide use for decades, and―more importantly―the sentence where I used that acronym was not essential in my message. The core of my message was its last 2 sentences. And of course, there is nothing preventing anyone from asking if they struggle to understand any part of a message.

> In your reply you argue that your message was "correct" (for some definition of the word).

I fail to see where I would have done that.

> That suggests to me that you may have missed my original point (my apologies, maybe I should have been clearer). And now, ironically, your communication style collided with mine 😉 .

Considering that you and I are both professional software engineers, I suppose our communication styles are not condemned to incompatibility. If you believe our communication styles are incompatible and think you have valuable information to communicate to me, I would suggest you explain which incompatibilities you perceive, and―more importantly―what can be done about it. My communication style is mostly defined by clarity and constructiveness. I am afraid that is not unique to me, but I do struggle to communicate with people who are unclear or not constructive.

**Comment by kees-jan** [comment]
Very well.

Over the course of this conversation, I've sent you several messages. Judging from your replies, you think I'm mostly wrong and have proceeded to explain to me why you think that is the case. You seem to not have considered the possibility that I might be in some sense correct (though maybe not as clear as you would have liked), and you have not asked follow-up questions. To me, that casts doubts on your inbound communication style compatibility.

Furthermore, I'm trying to provide feedback on your (admittedly outbound) communication style. And in that exact conversation, you praise your (admittedly inbound) communication style. To me, that's a sign we are not currently having a productive discussion. Additionally, that also tells me that we're having cultural differences. Doing something like that would be considered odd over here.

**Comment by Chealer** [comment]
Hi @kees-jan, and welcome to free software development,

> Very well.
> 
> Over the course of this conversation, I've sent you several messages. Judging from your replies, you think I'm mostly wrong and have proceeded to explain to me why you think that is the case. You seem to not have considered the possibility that I might be in some sense correct (though maybe not as clear as you would have liked), and you have not asked follow-up questions.

I don't see anyone asking follow-up questions here, at least about what you and I stated. If you think there should be questions about what you wrote, you are welcome to answer them.

Had I not considered the possibility that you might be partially correct, I would not have confirmed some of what you stated. However, it's important to consider that others don't have a monopoly on mistakes, and the Internet is not immune to the law of holes.

We certainly don't share the same outbound communication style; mine's diplomacy may unfortunately impair its clarity. So let me borrow your style's assertiveness and (alleged) verbosity for a moment.


> Furthermore, I'm trying to provide feedback on your (admittedly outbound) communication style. And in that exact conversation, you praise your (admittedly inbound) communication style.

I don't see where I would have done that...

> To me, that's a sign we are not currently having a productive discussion.

...but on that, I will not argue.

> Additionally, that also tells me that we're having cultural differences.

The opposite would be unlikely for 2 adults who hadn't communicated until last week. But to reiterate what I stated closer to your style: observing that without at least identifying specific differences has little to no value. What might be valuable would be concrete suggestions.

> Doing something like that would be considered odd over here.

I do not see what you mean by "that".

You seem to have a reasonable guess at who Workrave users are, so let me tell you about its developers. Most free software products which are entirely free and not distributed by any organization are mostly developed by philanthropists. Thus, [free software projects suffer from severe shortages of manpower](https://www.philippecloutier.com/Challenterests#Free_software).

Workrave does not have any commercial edition and is not marketed. At least I am not sponsored for any of my contributions, and my only personal gains from ITS triage are the modest chances each cleanup leads to a Workrave improvement which benefits me, and learning about Workrave's features. Therefore, when a senior colleague effectively donates some thousand euros and we're not fully satisfied, we try hard to focus on issues worse than teaching us our acronym of the day. If one contributes to the triage of tens of tickets which had been untriaged for years and advises you, doing no greater crime (except being―in your view―brief), please show some gratitude before suggesting they may be... unwelcoming.

No one is perfect, and free software communities are notable for failing to address poor quality and behavior. Criticism is vital, and quality criticism is often lacking even more pressingly in volunteerism-based free software, but as always, criticism should be thoughtful―focused, constructive and fair. We welcome neurodiversity and cannot require gratitude, but it's preferable for those with limited communication skills to leave dealing with subtle behavioral issues to more diplomatic colleagues.

Lastly, note that this ITS is most basic, and poor at sustaining complex discussions. Having a significant discussion which is only marginally related at best can derail a ticket. If you think you have something constructive to add, please do so in a discussion forum.

**Comment by rcaelers** [comment]
Hi @Chealer,

Thanks for taking the time to contribute to Workrave and to share your thoughts — I appreciate the engagement.

I don’t want to step into the discussion about “communication style compatibility,” but I do want to address one part of your message. The phrasing about “limited communication skills” and leaving certain matters to “more diplomatic colleagues” struck me as inappropriate and beyond what I consider acceptable in this project. Maybe it wasn’t intended personally, but it read that way in context.

@kees-jan’s involvement with Workrave goes back to the early days, before it was even publicly released, and his contributions have been essential. Seeing that remark directed at him felt unfair to me. I wanted to make that clear.

Just to clarify one other point: we use GitHub Issues not only for bug reports, but also for feature requests and improvement ideas. So requests without a reproducible “issue” are entirely valid here.

**Comment by pagin85** [comment]
Hi,
I am relatively unrelated to this entire discussion but wanted just to point out that is arrives in the inbox of people who are just interested in the "issue" / feature request.
If I can add any useful bit of information, this behaviour is part of the reason why I stopped using workrave. Since I did not use it since, I didn't update anything, so the version installed now is 10.10.48 - now on an updated Windows 11 (25H2), but then I probably still was on some version of windows 10.

**Comment by kees-jan** [comment]
Hi @pagin85,

I apologize for the noise. I triggered a discussion I felt was worth having, even though it eventually went nowhere, and I couldn't really think of a better place to have the conversation.

You're saying that you too would like to see a way to reset the restbreak timer, presumably? I feel it would make less sense for the micro-break and daily limit timers. The workarounds I describe of reducing your break time and/or clicking the "skip" button on the break window have been present since forever, so they do not depend on your workrave version.

Chealer was correct in pointing out that receiving "natural" break windows when the computer is locked needs to be enabled in preferences (Timer -> Rest break -> Start restbreak when screen is locked)

Are you now saying you were not aware of those workarounds, or are they not sufficient for you? And if I may ask: What, if anything, are you currently using and why? It always pays to check out the competition 😉 .

**Comment by Chealer** [comment]
Greetings Rob,

> Hi [@Chealer](https://github.com/Chealer),
> 
> Thanks for taking the time to contribute to Workrave and to share your thoughts — I appreciate the engagement.

My pleasure; the feeling is reciprocal.
 
> I don’t want to step into the discussion about “communication style compatibility,” but I do want to address one part of your message. The phrasing about “limited communication skills” and leaving certain matters to “more diplomatic colleagues” struck me as inappropriate and beyond what I consider acceptable in this project. Maybe it wasn’t intended personally, but it read that way in context.
> 
> [@kees-jan](https://github.com/kees-jan)’s involvement with Workrave goes back to the early days, before it was even publicly released, and his contributions have been essential. Seeing that remark directed at him felt unfair to me. I wanted to make that clear.

Hum. Apologies, but again, I allowed myself to borrow his communication style in the message you reacted to. Yet, I fail to see what would be unacceptable in the specific sentence you seem to react to, which sounds to me like common sense. I suppose you felt that he fit more in the "those with limited communication skills” part than in “more diplomatic colleagues”, but you are visibly in a much better position than me to tell how correct that is. You have contributed to Workrave a lot more than I did. I only know Kees from his contributions to a handful of Workrave tickets (mainly this one).

> Just to clarify one other point: we use GitHub Issues not only for bug reports, but also for feature requests and improvement ideas. So requests without a reproducible “issue” are entirely valid here.

GitHub Issues is intended for all kinds of issues, not just bugs, as previously discussed. I am guessing you simply did not read the discussion, which―since we are on the topic of discussion hygiene―may be a good reminder to mention [CBPP's ABCD](https://www.philippecloutier.com/blogpost136-CBPP-s-ABCD).

**Comment by Chealer** [comment]
Hi Philippe,
> Hi, I am relatively unrelated to this entire discussion but wanted just to point out that is arrives in the inbox of people who are just interested in the "issue" / feature request.

Yes, we are aware of that. Kees did express hesitancy about discussing this here from his first message.

> If I can add any useful bit of information, this behaviour is part of the reason why I stopped using workrave. Since I did not use it since, I didn't update anything, so the version installed now is 10.10.48 - now on an updated Windows 11 (25H2), but then I probably still was on some version of windows 10.

There is no such version, but you must mean 1.10.48.

-----

@kees-jan wrote:
> I apologize for the noise. I triggered a discussion I felt was worth having, even though it eventually went nowhere, and I couldn't really think of a better place to have the conversation.

A forum about development would have been a much better place. Failing that, this could have been discussed via a direct email.

**Comment by andrey0924** [comment]
> [@andrey0924](https://github.com/andrey0924) If your break was long enough to rest, and yet the cycle hasn't reset automatically, then maybe your break duration is set too high?
> 
> If your screen was locked during your break, then there should be a (natural) restbreak window visible when you return. If you press the "skip" button on it, the cycle should reset.

I usually work for 50 minutes, then take a 10-minute break.
But sometimes, I take a break outside of my schedule. For example, after 30 minutes of work, I take a fairly long break, after which I can work another 50 minutes without a break.

**In such cases, I currently have two options:**
1) Work for another 20 minutes, and then the break will automatically start for 10 minutes.
This option is ineffective, since I've already taken a break.
2) Or I can go into the settings and set the break timer to 1 second instead of 10 minutes, then start the break, and the timer will reset. But changing the settings there every time is a hassle.

My feature proposal describes a solution to this problem.

**Comment by andrey0924** [comment]
I've tried many other apps, but I haven't found anything better, and I'm still using Workrave.
Please implement this feature.

**Comment by Chealer** [comment]
@andrey0924: Please be specific. "a fairly long break" is too subjective. Ideally, provide a reproduction procedure. And again, at least specify which Workrave version you use when that occurs.

**Comment by andrey0924** [comment]
Now I'm using Workrave 1.10.54 for Windows.
"Quite a long break," for example, 1 hour for lunch or 30 min for something else.

**Comment by Chealer** [comment]
Thank you @andrey0924 
Which Windows version do you use? And can you share your timer configurations? In particular, after how long are micro-breaks and rest breaks set to start? A screenshot could do.

**Comment by andrey0924** [comment]
Now i'm using Windows 11 Pro,  version 25H2, buid 26200.7171
Timer configurations on screenshot
<img width="765" height="550" alt="Image" src="https://github.com/user-attachments/assets/df6b3d25-fe8b-467c-b375-7e0aae27f37a" />

**Comment by andrey0924** [comment]
I checked again, and now I can indeed reset the timer by clicking the "Start Break" button and then the "Skip breack" button.
However, there's no way to reset the microbreak.
The microbreak button isn't clickable, unlike the break button.
Maybe that's why I stopped using the microbreak feature. Because the time on the two counters starts to diverge.
And it becomes difficult to create a schedule based on the principle of
25 work + 5 breack + 25 work + 5 breack + 25 work + 15 breack

**Comment by andrey0924** [comment]
Screenshot of the timer after skipping a break
The break timer restarted at 1:00, but the microbreak timer did not reset (25 minutes).

<img width="91" height="75" alt="Image" src="https://github.com/user-attachments/assets/8c74c61d-ffbb-4a68-a8d2-cf03fa7e6f3f" />

**Comment by Chealer** [comment]
Thank you @andrey0924 

> However, there's no way to reset the microbreak. The microbreak button isn't clickable, unlike the break button.

Please describe or provide a screenshot of the button you refer to. Are you talking about a button displayed during a micro-break?

Is your Workrave in reading mode?

**Comment by kees-jan** [comment]
> Please describe or provide a screenshot of the button you refer to.

In the status window, if you click the cup icon in front of the restbreak progress bar, a restbreak is triggered. There is no similar functionality for microbreak or daily limit (nor should there be, in my opinion)

<img width="94" height="113" alt="Image" src="https://github.com/user-attachments/assets/e7a86fa4-4705-4dc7-a5ea-72e98f71a730" />

**Comment by andrey0924** [comment]
I use in Workrave Pomodoro Technique like this: https://en.wikipedia.org/wiki/Pomodoro_Technique
![Image](https://github.com/user-attachments/assets/b4186c4f-f800-4466-b725-5f88c6ea6496)

**For this technique i think we need in Workrave settings like this:**
![Image](https://github.com/user-attachments/assets/7d553621-20ab-4ea1-8551-ed9f5f74748e)
![Image](https://github.com/user-attachments/assets/f58700d9-5f09-4c10-8db6-d55db78d5e34)

**But we need this new feautures in Workrave:**
1) On screenshot Step 6 to 2 in Wiki: "After four pomodori are done, take a long break (typically 20 to 30 minutes) instead of a short break. Once the long break is finished, Set the Pomodoro timer (typically for 25 minutes)". 
**_It's mean that we need in Workrave to reset automaticaly short timer (25 min) after long "Rest break"._** (only the long timer is currently updated)

2) And this buttons I think will be helpful for using in WorkRave
https://github.com/schmich/marinara/blob/master/assets/screenshots/menu.png

**_Restart Timer_**
- Start Focusing (working cycle) - It's mean reset Short and long timer
- Start Short Break
- Start Long Break - we already have this button, if you click the cup icon

**Comment by Chealer** [comment]
> > Please describe or provide a screenshot of the button you refer to.
> 
> In the status window, if you click the cup icon in front of the restbreak progress bar, a restbreak is triggered. There is no similar functionality for microbreak or daily limit (nor should there be, in my opinion)

Ooooh! Thank you Kees, I see. That is a nice feature, but **NOT** obvious.
I think similar functionality for micro-breaks (or at least, "small breaks") would make sense.

**Comment by Chealer** [comment]
Thank you @andrey0924, but I struggle to reconcile your latest comment with [your first comment](https://github.com/rcaelers/workrave/issues/335#issuecomment-3608421604). Do you really consider that you use the Pomodoro Technique? And if so, with which durations?

And again, is your Workrave in reading mode?

**Comment by andrey0924** [comment]
I created this task four years ago when I tried using Workrave for Pomidoro, but it didn't work for the reasons described above.
I've since tried many different programs and occasionally returned to yours, but overall, I haven't found anything better than yours, as it includes various exercises during breaks that somewhat help me avoid being glued to my smartphone and generally increase my productivity. Now I only use your software with long breaks after 50 minutes or 1 hour 10 minutes of work. Although, I'd certainly like to use the Pomidoro technique.

I also have an idea to expand the list of exercises and update the images to something not from the 90s, but that's another story.

**Comment by Chealer** [comment]
> I created this task four years ago when I tried using Workrave for Pomidoro, but it didn't work for the reasons described above. I've since tried many different programs and occasionally returned to yours, but overall, I haven't found anything better than yours, as it includes various exercises during breaks that somewhat help me avoid being glued to my smartphone and generally increase my productivity. Now I only use your software with long breaks after 50 minutes or 1 hour 10 minutes of work. Although, I'd certainly like to use the Pomidoro technique.

Ah, thank you

Sorry for repeating myself, but is your Workrave set to reading mode?

> I also have an idea to expand the list of exercises and update the images to something not from the 90s, but that's another story.

Indeed... we are already having a hard enough time keeping this ticket focused, so feel free to share, but not in this ticket.

**Comment by andrey0924** [comment]
Right now, reading mode is turned off, but I turn it on from time to time if I'm using the computer, for example, to watch video courses. I might leave reading mode on, because if I'm at the computer and the timer is on, it means I'm doing something productive. In short, this feature isn't particularly useful for me; that is, I don't need to pause the timer if I'm not moving the mouse or typing.

**Comment by andrey0924** [comment]
I suggest at least implementing what's described in this ticket to start.
If you'd like to implement a full-fledged Pomodoro mode, I can describe how that could be done in a separate ticket.
Pomodoro mode is very popular, in my opinion, and if this feature weren't hidden away in the settings, but rather prominently featured, the app's target audience would expand significantly.
In any case, everything described in this ticket will be needed to implement a full-fledged Pomodoro mode.
In short, a couple more things will be needed for a full-fledged Pomodoro mode, and the app could be called "Workrave & Pomodoro" )))

**Comment by Chealer** [comment]
Thank you Andrei

At this point we are only trying to understand your issue, so it would help to keep solutions for later (Pomodoro discussion is surely even better in an entirely different ticket).

If I understand correctly:
1. You have Workrave 1.10.54 on Windows 11 Pro 25H2 set with:
   1. Rest breaks of 10 minutes after every 50 minutes of work
   2. No micro-breaks
   3. **Reading mode is disabled**.
2. At 5:00, you work, being constantly active on your PC for 30 minutes.
3. At 5:30, you take a break for 45 minutes.
4. At 6:15, you go back to your PC and work constantly.

In such a situation, you are saying that Workrave would recommend you a rest break at 6:35. Is that correct?

**Comment by andrey0924** [comment]
I made this ticket when I tried to use Workrave for Pomidoro, and I still want it. I want to work in this shuddle:
- Work 25 min
- Micro break 5 min
- Work 25 min
- Micro break 5 min
- Work 25 min
- Long break 30 min
- Work 25 min
- Micro break 5 min
...

**Comment by andrey0924** [comment]
If you try working on this schedule, the 25-minute timer won't reset after a long 30-minute break. So, during work, I might end up taking micro-breaks late, postponing them a couple of times, which is always inevitable, as we're not robots.

**Comment by andrey0924** [comment]
Settings for Pomidoro (3 rouns of 25 min work, then 30 min of break):
- I have Workrave 1.10.54 on Windows 11 Pro 25H2 set with:
- No micro-breaks
- Reading mode is disabled or enabled (for me it's doesn't meter)
- Micro break 25 min work, 5 min rest
- Long break 1 hour 30 min of work, 30 min rest

**Comment by Chealer** [comment]
I am unsubscribing from this ticket and giving up on trying to understand what it reports. My best guess is documented in [my previous comment](https://github.com/rcaelers/workrave/issues/335#issuecomment-3620446663). In the absence of improvement, I recommend marking this report as unclear.

<details>
  <summary>License</summary>
This comment and any other from myself on this ticket is offered under the terms of CC0 1.0.
</details>

**Comment by andrey0924** [comment]
I'm also unsubscribing. A simple function (resetting a timer) requires so much discussion and ultimately yields zero results.
